In [50]:
import pandas as pd
import random
from pathlib import Path
import scipy.stats as st
from math import ceil

In [42]:
# Load breath-by-breath articles
bbb_df_path = Path('/Users/antonhesse/Desktop/Anton/Education/UMN/Lab and Research/HSPL/CPET_scoping_review/data/cpet_articles/text_analysis/bbb_articles.csv')
bbb_df = pd.read_csv(bbb_df_path)

In [43]:
# Load known mixing chamber and Douglas bag articles
mc_db_df_path = Path('/Users/antonhesse/Desktop/Anton/Education/UMN/Lab and Research/HSPL/CPET_scoping_review/data/cpet_articles/text_analysis/Mixing Chamber Douglas Bag - Sheet1.csv')
mc_db_df = pd.read_csv(mc_db_df_path)

In [44]:
# create data frame of articles known to use mixing chambers or Douglas bag
non_bbb = mc_db_df[mc_db_df['bbb'] == False]

In [45]:
# remove mixing chamber and Douglas bag articles from breath-by-breath data frame
bbb_df_cleaned = bbb_df[~bbb_df['doi_suffix'].isin(non_bbb['doi_suffix'].to_list())].reset_index(drop=True)

In [52]:
# Thanks Dr. Zieffler and your video to Khan Academy
# https://www.khanacademy.org/math/ap-statistics/xfb5d8e68:inference-categorical-proportions/one-sample-z-interval-proportion/v/determining-sample-size-based-on-confidence-and-margin-of-error
margin_of_error = 0.03
z = st.norm.ppf(0.975)
p = 0.5

n = ceil(((margin_of_error / z)**2)**-1 * p * (1 - p))
n


1068

Although the minimum number or articles required to obtain a margin of error of ±3% with 95% confidence is 1068, we will round up to 1100 because we predict to find a small number of ineligible articles. Although we have eliminated the majority of ineligible articles already, we expect to find a small number of revierw articles, animal studies, or studies that used systems with mixing chambers or Dougalas bags.

In [53]:
n_rounded_up = 1100
random_bbb_df = bbb_df_cleaned.sample(n_rounded_up, random_state=34789).reset_index(drop=True)
random_bbb_df

doi_suffix  \
0                     phy2.13945   
1           02640414.2014.926384   
2              s00421-004-1045-1   
3                ijerph181910426   
4                     phy2.14624   
...                          ...   
1095                  bf00846030   
1096              sj.ijo.0803135   
1097           s40618-015-0411-7   
1098  01.mss.0000135977.12456.8f   
1099        mss.0000000000002488   

                                                   path    bbb  oxycon  \
0     /Users/antonhesse/Desktop/Anton/Education/UMN/...   True   False   
1     /Users/antonhesse/Desktop/Anton/Education/UMN/...  False    True   
2     /Users/antonhesse/Desktop/Anton/Education/UMN/...   True   False   
3     /Users/antonhesse/Desktop/Anton/Education/UMN/...  False   False   
4     /Users/antonhesse/Desktop/Anton/Education/UMN/...   True    True   
...                                                 ...    ...     ...   
1095  /Users/antonhesse/Desktop/Anton/Education/UMN/...   True   False   
1096  /Users/antonhesse/Desktop/Anton/Education/UMN/...   True   False   
1097  /Users/antonhesse/Desktop/Anton/Education/UMN/...   True   False   
1098  /Users/antonhesse/Desktop/Anton/Education/UMN/...   True   False   
1099  /Users/antonhesse/Desktop/Anton/Education/UMN/...   True   False   

      carefusion  medgraphics  sensormedics  cosmed  minato  pred_bbb  ...  \
0          False        False         False   False   False      True  ...   
1          False        False         False   False   False      True  ...   
2          False        False         False    True   False      True  ...   
3          False        False          True   False   False      True  ...   
4          False        False         False   False   False      True  ...   
...          ...          ...           ...     ...     ...       ...  ...   
1095       False        False         False   False   False      True  ...   
1096       False        False         False   False   False      True  ...   
1097       False        False         False   False   False      True  ...   
1098       False         True         False   False   False      True  ...   
1099       False        False         False    True   False      True  ...   

          first_oa_location.url_for_landing_page  \
0            https://doi.org/10.14814/phy2.13945   
1                                            NaN   
2                                            NaN   
3     https://dx.doi.org/10.3390/ijerph181910426   
4            https://doi.org/10.14814/phy2.14624   
...                                          ...   
1095                                         NaN   
1096      https://doi.org/10.1038/sj.ijo.0803135   
1097                                         NaN   
1098                                         NaN   
1099                                         NaN   

                 first_oa_location.evidence first_oa_location.license  \
0                     oa journal (via doaj)                     cc-by   
1                                       NaN                       NaN   
2                                       NaN                       NaN   
3     oa repository (via OAI-PMH doi match)                     cc-by   
4                     oa journal (via doaj)                     cc-by   
...                                     ...                       ...   
1095                                    NaN                       NaN   
1096                    open (via free pdf)                       NaN   
1097                                    NaN                       NaN   
1098                                    NaN                       NaN   
1099                                    NaN                       NaN   

     first_oa_location.version  first_oa_location.host_type  \
0             publishedVersion                    publisher   
1                          NaN                          NaN   
2                          NaN                          NaN   
3        

In [39]:
random_bbb_df['doi_suffix'].to_csv('/Users/antonhesse/Desktop/Anton/Education/UMN/Lab and Research/HSPL/CPET_scoping_review/data/cpet_articles/text_analysis/random_avg_articles.csv', index=False)

In [40]:
random_bbb_df['doi_suffix'].to_clipboard(index=False)